File: TBSWeb/ClientApp/src/controller/AccountController.js
import axios from 'axios';

export async function getListAccounts () {
    try {
        const { data } = await axios.get('/accounts');

        data.forEach(acc => {
            acc.serverUrl = acc.serverUrl.replace(/(^\w+:|^)\/\//, '');
        });
        return data;
    } catch (e) {
        console.log(e);
    }
    return null;
}

export async function getAccount (index) {
    try {
        const { data } = await axios.get(`/accounts/${index}`);
        data.serverUrl = data.serverUrl.replace(/(^\w+:|^)\/\//, '');
        return data;
    } catch (e) {
        console.log(e);
    }
    return null;
}

export async function addAccount (data) {
    try {
        await axios.post('/accounts', data);
    } catch (e) {
        console.log(e);
    }
}

export async function editAccount (index, data) {
    try {
        await axios.put(`/accounts/${index}`, data);
    } catch (e) {
        console.log(e);
    }
}

export async function deleteAccount (index) {
    try {
        await axios.delete(`/accounts/${index}`);
    } catch (e) {
        console.log(e);
    }
}

export async function getTasks (index) {
    try {
        console.log(index);
        const { data } = await axios.get(`/debug/tasks/${index}`);
        console.log(data);
        return data;
    } catch (e) {
        console.log(e);
    }
    return null;
}
const current = {
    account: -1,
    village: -1,
};

Object.seal(current);
export { current };

File: TBSReact/Controllers/Account/AccountController.cs
using Microsoft.AspNetCore.Mvc;
using TbsReact.Models;
using TbsReact.Singleton;
using TbsReact.Extension;
using System.Collections.Generic;
using TbsCore.Database;
using System;

namespace TbsReact.Controllers
{
    [ApiController]
    [Route("accounts")]
    public class AccountController : ControllerBase
    {
        [HttpGet]
        public ActionResult<List<Account>> GetAccounts()
        {
            var AccountInfoList = new List<Account>();
            for (int i = 0; i < AccountData.Accounts.Count; i++)
            {
                AccountInfoList.Add(AccountData.Accounts[i]);
            }

            return AccountInfoList;
        }

        [HttpGet("{index:int}")]
        public ActionResult<Account> GetAccount(int index)
        {
            var acc = AccountData.GetAccount(index);
            if (acc == null)
            {
                return NotFound();
            }
            return acc;
        }

        [HttpPost]
        public ActionResult AddAccount([FromBody] NewAccount data)
        {
            var account = data.Account;
            var accesses = data.Accesses;
            if (string.IsNullOrEmpty(account.Name)
                || string.IsNullOrEmpty(account.ServerUrl))
            {
                return BadRequest();
            }
            var acc = account.GetAccount(accesses);
            DbRepository.SaveAccount(acc);
            AccountManager.Accounts.Add(acc);
            AccountData.AddAccount(account);

            return Ok();
        }

        [HttpPatch("{index:int}")]
        public ActionResult EditAccount(int index, [FromBody] NewAccount data)
        {
            var account = data.Account;
            var accesses = data.Accesses;

            if (string.IsNullOrEmpty(account.Name) ||
                string.IsNullOrEmpty(account.ServerUrl))
            {
                return BadRequest();
            }
            var accountOld = AccountData.GetAccount(index);
            if (accountOld == null)
            {
                return NotFound();
            }

            var acc = AccountManager.GetAccount(accountOld);

            AccountData.EditAccount(index, account);

            acc.AccInfo.Nickname = account.Name;
            acc.AccInfo.ServerUrl = account.ServerUrl;

            acc.Access.AllAccess.Clear();
            foreach (var access in accesses)
            {
                acc.Access.AllAccess.Add(new TbsCore.Models.Access.Access
                {
                    Password = access.Password,
                    Proxy = access.Proxy.Ip,
                    ProxyPort = access.Proxy.Port,
                    ProxyUsername = access.Proxy.Username,
                    ProxyPassword = access.Proxy.Password,
                });
            }
            DbRepository.SaveAccount(acc);
            return Ok();
        }

        [HttpDelete("{index:int}")]
        public ActionResult DeleteAccount(int index)
        {
            var account = AccountData.GetAccount(index);
            if (account == null)
            {
                return NotFound();
            }
            var acc = AccountManager.GetAccount(account);

            DbRepository.RemoveAccount(acc);
            AccountManager.Accounts.Remove(acc);
            AccountData.DeleteAccount(index);

            return Ok();
        }
    }
}